packages/teams-js/test/public/pages.spec.ts

```diff
@@ -210,42 +335,917 @@ describe('AppSDK-TeamsAPIs', () => {
     });
   });
 
-  describe('returnFocus', () => {
-    it('should successfully returnFocus', async () => {
-      await utils.initializeWithContext(FrameContexts.content);
+  describe('Testing pages.navigateToApp function', () => {
+    const navigateToAppParams: pages.NavigateToAppParams = {
+      appId: 'fe4a8eba-2a31-4737-8e33-e5fae6fee194',
+      pageId: 'tasklist123',
+      webUrl: 'https://tasklist.example.com/123',
+      channelId: '19:cbe3683f25094106b826c9cada3afbe0@thread.skype',
+      subPageId: 'task456',
+    };
+
+    it('pages.navigateToApp should not allow calls before initialization', async () => {
+      await expect(pages.navigateToApp(navigateToAppParams)).rejects.toThrowError(
+        'The library has not yet been initialized',
+      );
+    });
+
+    const allowedContexts = [
+      FrameContexts.content,
+      FrameContexts.sidePanel,
+      FrameContexts.settings,
+      FrameContexts.task,
+      FrameContexts.stage,
+      FrameContexts.meetingStage,
+    ];
+
+    Object.keys(FrameContexts).forEach(context => {
+      if (allowedContexts.some(allowedContext => allowedContext === context)) {
+        it(`pages.navigateToApp should allow calls from ${context} context`, async () => {
+          expect.assertions(1);
+          await utils.initializeWithContext(context);
+          utils.setRuntimeConfig({ apiVersion: 1, supports: { pages: {} } });
+
+          const promise = pages.navigateToApp(navigateToAppParams);
+
+          const navigateToAppMessage = utils.findMessageByFunc('pages.navigateToApp');
+          utils.respondToMessage(navigateToAppMessage, true);
+
+          await expect(promise).resolves.toBe(undefined);
+        });
+
+        it('pages.navigateToApp should successfully send the navigateToApp message', async () => {
+          await utils.initializeWithContext(context);
+          utils.setRuntimeConfig({ apiVersion: 1, supports: { pages: {} } });
+
+          const promise = pages.navigateToApp(navigateToAppParams);
+
+          const navigateToAppMessage = utils.findMessageByFunc('pages.navigateToApp');
+          utils.respondToMessage(navigateToAppMessage, true);
+          await promise;
+
+          expect(navigateToAppMessage).not.toBeNull();
+          expect(navigateToAppMessage.args[0]).toStrictEqual(navigateToAppParams);
+        });
+
+        it('pages.navigateToApp should successfully send an executeDeepLink message for legacy teams clients', async () => {
+          await utils.initializeWithContext(context);
+
+          const promise = pages.navigateToApp(navigateToAppParams);
```

Comment:
Suggest setting the runtime.isLegacyTeams to true so the test is more explicit and not relying on default behavior. Although actually it might be good to add a test that validates the default behavior as well, in case it ever changes.

Reply:
We already have the test for default behavior, added the `isLegacy` to true in latest commit.